lib/DxilValidation/DxilValidation.cpp

```diff
@@ -970,6 +970,169 @@ static void ValidateImmOperandForMathDxilOp(CallInst *CI, DXIL::OpCode opcode,
   }
 }
 
+static bool CheckFromRegisterInterpretations(uint32_t input) {
+  DXIL::ComponentType ValidSet[] = {
+      DXIL::ComponentType::I16,         DXIL::ComponentType::U16,
+      DXIL::ComponentType::I32,         DXIL::ComponentType::U32,
+      DXIL::ComponentType::F16,         DXIL::ComponentType::F32,
+      DXIL::ComponentType::PackedS8x32, DXIL::ComponentType::PackedU8x32,
+      DXIL::ComponentType::U8,          DXIL::ComponentType::I8,
+      DXIL::ComponentType::F8_E4M3,     DXIL::ComponentType::F8_E5M2};
+
+  for (auto Val : ValidSet) {
+    if (Val == static_cast<DXIL::ComponentType>(input))
+      return true;
+  }
+  return false;
+}
+
+static bool CheckInMemoryInterpretations(uint32_t input) {
+  DXIL::ComponentType ValidSet[] = {
+      DXIL::ComponentType::I16,     DXIL::ComponentType::U16,
+      DXIL::ComponentType::I32,     DXIL::ComponentType::U32,
+      DXIL::ComponentType::F16,     DXIL::ComponentType::F32,
+      DXIL::ComponentType::U8,      DXIL::ComponentType::I8,
+      DXIL::ComponentType::F8_E4M3, DXIL::ComponentType::F8_E5M2};
+
+  for (auto Val : ValidSet) {
+    if (Val == static_cast<DXIL::ComponentType>(input))
+      return true;
+  }
+  return false;
+}
+
+static bool CheckMatrixLayout(unsigned input) {
+  return (input <=
+          static_cast<unsigned>(DXIL::DXILMatrixLayout::OuterProductOptimal));
+}
+
+static void ValidateImmOperandsForMatVecOps(CallInst *CI, DXIL::OpCode opcode,
+                                            ValidationContext &ValCtx) {
+
+  // Check Common operands
+  llvm::Value *InputIsUnsigned =
+      CI->getOperand(DXIL::OperandIndex::kMatVecMulIsInputUnsignedIdx);
+  llvm::Value *InputInterpretation =
+      CI->getOperand(DXIL::OperandIndex::kMatVecMulInputInterpretationIdx);
+  llvm::Value *MatrixInterpretation =
+      CI->getOperand(DXIL::OperandIndex::kMatVecMulMatrixInterpretationIdx);
+  llvm::Value *MatrixM =
+      CI->getOperand(DXIL::OperandIndex::kMatVecMulMatrixMIdx);
+  llvm::Value *MatrixK =
+      CI->getOperand(DXIL::OperandIndex::kMatVecMulMatrixKIdx);
+  llvm::Value *MatrixLayout =
+      CI->getOperand(DXIL::OperandIndex::kMatVecMulMatrixLayoutIdx);
+  llvm::Value *MatrixTranspose =
+      CI->getOperand(DXIL::OperandIndex::kMatVecMulMatrixTransposeIdx);
+
+  if (!llvm::isa<llvm::Constant>(InputIsUnsigned)) {
+    ValCtx.EmitInstrError(CI,
+                          ValidationRule::InstrMatVecOpIsUnsignedFlagsAreConst);
+  }
+
+  if (!llvm::isa<llvm::Constant>(InputInterpretation) ||
+      !llvm::isa<llvm::Constant>(MatrixInterpretation)) {
+    ValCtx.EmitInstrError(
+        CI, ValidationRule::InstrLinalgInterpretationParamAreConst);
+  }
+
+  // Check if InputInterpretation and MatrixInterpretation are valid
+  ConstantInt *Ii = cast<ConstantInt>(InputInterpretation);
+  auto IiValue = Ii->getLimitedValue();
+  if (!CheckFromRegisterInterpretations(IiValue)) {
```

Comment:
Coding standards:
Since these are initials, they should both be capitalized.
```suggestion
  ConstantInt *II = cast<ConstantInt>(InputInterpretation);
  auto IIValue = II->getLimitedValue();
  if (!CheckFromRegisterInterpretations(IIValue)) {
```
Same with others below - `MI`, `ML`, `BI`.